### PR TITLE
Maria DB support

### LINKED ISSUE #471: tests+dialect for MariaDB
Testcontainers has [a module for MariaDB](https://www.testcontainers.org/modules/databases/mariadb/).

MariaDB is for the most part just a flavor of MySQL, and we support it via the Vert.x driver for MySQL. But I think that's not quite good enough. There are some (very) minor differences in the Hibernate `Dialect`, and so we should treat it as a separate case.

So we should set up tests to run on Maria, and make sure we detect it and use the correct dialect by default, instead of making the user specify it explicitly.

### PR DESCRIPTION
**Pull request by gavinking**
Fixes #471.

### CODE REVIEW

**Comment by gavinking** [comment]
Couple of very minor things to clean up before I merge.

**Comment by gavinking** [comment]
Oh, and we need to set up the tests to run on the CI. @DavideD can you make that happen?

**Comment by DavideD** [comment]
Sure, I will do it when we have merged this

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/EagerManyToOneAssociationTest.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public void manyToOneIsNull(TestContext context) {
 	@DiscriminatorValue("N")
 	//@Inheritance(strategy = InheritanceType.JOINED)
 	public static class Book {
-		public static final String TABLE = "Book";
+		public static final String TABLE = "Bookz";
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/EagerManyToOneAssociationTest.java:104]: Just a curiosity, how come you also need to change the table name?

Comment (gavinking) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/EagerManyToOneAssociationTest.java:104]: Well MariaDB seems to be a lot pickier about that stuff for some reason I wasn't able to fully nail down. All I know is I wish it supported `cascade`.

Comment (gavinking) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/EagerManyToOneAssociationTest.java:104]: (In `drop table` I mean.)

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/containers/VertxMariaContainer.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import org.testcontainers.containers.MariaDBContainer;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Normally the MySQLContainer checks for readiness by attempting to establish a JDBC connection
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/containers/VertxMariaContainer.java:21]: s/`MySQLContainer`/`MariaDBContainer`/

Comment (gavinking) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/containers/VertxMariaContainer.java]: Yup, already noticed.

**Comment by gavinking** [comment]
> Sure, I will do it when we have merged this

OK, I'll merge it now then.

**Pull request merged**